WalletWasabi/Services/WalletService.cs

```diff
@@ -386,9 +386,10 @@ private void ProcessTransaction(SmartTransaction tx, List<HdPubKey> keys = null)
 			if (tx.Height.Type == HeightType.Chain)
 			{
 				MemPool.TransactionHashes.TryRemove(txId); // If we have in mempool, remove.
-				SmartTransaction foundTx = TransactionCache.FirstOrDefault(x => x == tx); // If we have in cache, update height.
-				if (foundTx != default(SmartTransaction))
+				var found = TransactionCache.Contains(tx); // If we have in cache, update height.
```

Comment:
@lontivero You cannot do this for concurrent collections.

Reply:
Nor I think it adds any speed, I think it even worsens it, assuming the FirstOrDefault and Contains are the same speed.

Reply:
`Contains` introduces a really big improvement, I measured it. This is also clear if we review the ConcurrentHashSet code, it goes directly to the bucket that contains the element, after all, it is a hashset.

Reply:
@lontivero Yes, I measured it, too, but I guess you'll get to those comments of mine soon: https://github.com/zkSNACKs/WalletWasabi/pull/702#issuecomment-426619485